todo tests for GH 16364 (array element lazily created at wrong index)

1-1sam opened the pull request:
This PR adds some todo tests for #16364 that test the following behavior:
* An array element that was previously uninitialized is created at the correct index when assigned from a subroutine's @_ alias which modifies the element's original array. The element is within the array.
* The same as the above but the element is outside of the array's range.
* An array element that was previously uninitialized is created at the correct index when assigned from a foreach loop's $_ alias which modifies the element's original array.

The first test seems to pass, the other two do not.

-------------------------------------------------------------------------------
* This set of changes does not require a perldelta entry.

khwilliamson commented on t/run/todo.t:
```diff
@@ -137,6 +137,38 @@ TODO: {
+    @arr = ();
+    $arr[1] = 1;
+    foo($arr[0]);
+    is($arr[1], 3, 'Array element created at correct index from subroutine @_ alias');
```
The test name should include something like "within range" and the GH number

khwilliamson commented on t/run/todo.t:
```diff
@@ -137,6 +137,38 @@ TODO: {
+    @arr = ();
+    $arr[1] = 1;
+    foo($arr[5]);
+    is($arr[6], 3, 'Array element created at correct index from subroutine @_ alias');
```
The test name should include something like "outside range" and the GH number

khwilliamson commented on t/run/todo.t:
```diff
@@ -137,6 +137,38 @@ TODO: {
+        $_ = 3;
+        last;
+    }
+    is($arr[1], 3, 'Array element created at correct index from foreach $_ alias');
```
The test name should include  the GH number

khwilliamson reviewed (dismissed):
This looks good to me except the test names should be distinguishable from each other.

And we discovered that when we move passing tests to an appropriate .t that their ticket number gets lost.  So we are now asking to put the number in each test name as well.  I'll shortly update the comments at the beginning of the file to say that.  But here's one possibility for the name for your first test

'Array element created at correct index from subroutine @_ alias: GH 16364'

1-1sam committed:
todo tests for GH 16364 (array element lazily created at wrong index)

This commit adds some todo tests for GH 16364 that test the following
behavior:
* An array element that was previously uninitialized is created at the
  correct index when assigned from a subroutine's @_ alias which
  modifies the element's original array. The element is within the
  array.
* The same as the above but the element is outside of the array's
  range.
* An array element that was previously uninitialized is created at the
  correct index when assigned from a foreach loop's $_ alias which
  modifies the element's original array.
